server/src/banner.js

```diff
@@ -0,0 +1,25 @@
+function define(modules, cb) {
+  function parseStringArray(argName) {
+    const index = process.argv.indexOf(argName);
+    if (index < 0 || index === process.argv.length - 1) {
+        return [];
+    }
+    const arg = process.argv[index + 1];
+    return arg.split(',');
+  }
+  const TSSERVER = 'typescript/lib/tsserverlibrary';
+  let tsserverPath;
+  try {
+    tsserverPath = require.resolve(TSSERVER, {
+      paths: parseStringArray('--typeScriptProbeLocations'),
+    });
+  }
+  catch {}
+  const resolvedModules = modules.map(m => {
+    if (m === 'typescript') {
+      throw new Error(`'typescript' should never be used. Use '${TSSERVER}' instead.`)
+    }
+    return require(m === TSSERVER && tsserverPath || m);
```

Comment:
Can you pull this out in a separate/longer conditional to more clearly explain what is being loaded? It took me a minute to realize this is equivalent to `m === TSSERVER && tsserverPath ? tsserverPath : m`.

Reply:
done.